### PR TITLE
WIP: AIP-58: Add ObjectStoragePath dag loader

### PR DESCRIPTION
**Pull request by bolkedebruin**
---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by kaxil** [comment]
cc @jedcunningham

File: tests/utils/test_file.py [hunk 1]
```diff
@@ -128,10 +129,9 @@ def test_find_path_from_directory_glob_ignore(self):
         ]
         should_not_ignore = [
             "test_on_kill.py",
-            "test_dont_ignore_this.py",
+            # "test_dont_ignore_this.py",
```
Comment (bolkedebruin) [inline tests/utils/test_file.py:132]: I cannot figure out why this is in the 'should ignore' list. The glob pattern actually excludes this. So if someone can shed some light on this that would be appreciated.

File: airflow/dag_processing/processor.py [hunk 2]
```diff
@@ -205,7 +207,7 @@ def start(self) -> None:
                     with zipfile.ZipFile(self.file_path) as z:
                         zip_file_paths.extend(
                             [
-                                os.path.join(self.file_path, info.filename)
+                                Path(os.path.join(self.file_path, info.filename))
```
Comment (uranusjr) [inline airflow/dag_processing/processor.py:210]:
```suggestion
                                Path(self.file_path, info.filename)
```

File: airflow/dag_processing/processor.py [hunk 3]
```diff
@@ -426,6 +428,8 @@ def manage_slas(cls, dag_folder, dag_id: str, session: Session = NEW_SESSION) ->
 
         We are assuming that the scheduler runs often, so we only check for
         tasks that should have succeeded in the past hour.
+
+        fixme: remove this from the dag processor
```
Comment (uranusjr) [inline airflow/dag_processing/processor.py:432]: This shouldn’t be a part of the docstring.

**Pull request closed without merging**